howoo/문자열 내 마음대로 정렬하기.js

```diff
@@ -0,0 +1,16 @@
+  function solution(strings, n) {
+    const result = strings
+      .sort() // 알파벳정렬
+      .map((string) => ({ string: string, nThLetter: string[n] })) // 원래 단어와 단어의 n번째 알파벳을 객채의 배열로 맵핑
+      .sort((a, b) => {
+        if (a.nThLetter < b.nThLetter) {
+          return -1;
+        }
+        if (a.nThLetter > b.nThLetter) {
+          return 1;
+        }
+        return 0;
+      }) // nThLetter 알파벳순으로 정렬
+      .map((obj) => obj.string); // 원래 단어만 매핑
```

Comment:
```suggestion
      .sort() // 알파벳 정렬 
      .sort((a, b) => { // n 인덱스 기준으로 정렬
        if (a[n] < b[n]) return -1;
        if (a[n] > b[n]) return 1;
        return 0;
      })
```

기존 단어와 n번 째 알파벳을 매핑하지 않고, `a[n]`과 같이 직접적으로 n번 째 알파벳 기준으로 정렬한다면 2회의 매핑 과정이 필요없을 것 같은데 어떻게 생각하시나요?

Reply:
오! 맞네요.
감사합니다!